.circleci/config.yml

```diff
@@ -14,37 +18,64 @@ jobs:
         command:
           - server
           - data
+  docker_builder:
+    working_directory: ~/project
+    docker:
+      - image: circleci/golang
+
+commands:
+  install_base:
     steps:
       - run:
-          name: Install packages
+          name: Install packages with gcloud SDK installed locally
           command: |
-            sudo apt-get update -y
-            sudo apt-get install lsb-release
+            sudo apt-get update -y --quiet
+            sudo apt-get install --quiet lsb-release
             export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)"
             echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
             curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add
-            sudo apt-get update -y
-            sudo apt-get install -y git build-essential
-            sudo apt-get install -y google-cloud-sdk
-            sudo apt-get install -y shellcheck
+            sudo apt-get update -y --quiet
+            sudo apt-get install -y --quiet git build-essential google-cloud-sdk shellcheck zsh
+
+  install_tools:
+    steps:
       - run:
-          name: Install go tools
+          name: Install go tools for testing
+          environment:
+            GO111MODULE: 'on'
           command: |
-            go get -u gotest.tools/gotestsum
-            go get -u github.com/matryer/moq
-            go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
-            mkdir -p /tmp/test-results
+            go get -u gotest.tools/gotestsum@latest
+            go get -u github.com/matryer/moq@latest
+            go get -u github.com/golangci/golangci-lint/cmd/golangci-lint@latest
+            go clean -modcache
+
+  login_to_google:
+    steps:
+      - run:
+          name: Initialize gcloud SDK
+          command: |
+            echo $GCLOUD_SERVICE_KEY | gcloud auth activate-service-account --key-file=-
```

Comment:
note to self: this environment variable is [set through the circleci web ui](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project)